andela/ah-the-jedi-backend
# #165305267 users should be able to rate an article

### What does this PR do?
Implements rate an article functionality

### Description of Task(s) to be completed?
- ensure user can only rate articles authored by other authors
- ensure only authenticated users can rate an article
- calculate the average rating of an article
- implement functionality to update rating if user has already rated an article
- include the average rating of an article in the object returned by an article
- add unit tests

### How should this be manually tested?

Clone this repository on local machine and CD into project folder
`$ git clone git@github.com:andela/ah-the-jedi-backend.git`
`$ cd ah-the-jedi-backend`

Check out to the PR's branch
`$ git checkout ft-rate-an-article-165305267`

Create and activate virtual environment
`$ python3 -m venv venv`
`$ source .venv/bin/activate`

Install app dependencies from requirements.txt
`$ pip install -r api/requirements.txt`

Make migrations and run the server
`$ python manage.py makemigrations`
`$ python manage.py migrate`
`$ python manage.py runserver`

**Use postman to test the endpoints**


           POST <your-localhost>/api/users/



       POST <your-localhost>/api/login/ 

       Use the token provided after login for authorization 

- [x] **rate an article**

      `POST`  `<you-local-host>/api/articles/<slug>/rate`

- [x] **view average rating**

      `GET`  `<you-local-host>/api/articles/<slug>/`


### What are the relevant PT stories?
[#165305267](https://www.pivotaltracker.com/story/show/165305267)

### Screenshots

**Rate article**
![image](https://user-images.githubusercontent.com/11965186/57289456-12092180-70c4-11e9-8c62-6f557ef5f94f.png)

**View average rating**
![image](https://user-images.githubusercontent.com/11965186/57289699-82b03e00-70c4-11e9-9e3c-3496239586cb.png)
